pandas/tests/series/methods/test_replace.py

```diff
@@ -799,3 +799,15 @@ def test_replace_numeric_column_with_na(self, val):
 
         ser.replace(to_replace=1, value=pd.NA, inplace=True)
         tm.assert_series_equal(ser, expected)
+
+    def test_replace_ea_float_with_bool(self):
```

Comment:
Can we use the parametrize decorator to test more branches of the logic tree in `pandas/core/missing.py`::Ln104-131?

Reply:
Why would this be necessary? everything below l116 is existing logic that is already covered by tests

Reply:
Ok, what about the three logic branches from 104 to 117 in `pandas/core/missing.py` - are they all covered? Do cases go into the expected branch?

Reply:
yes that's what the new tests are for

Reply:
Ok, and testing series of only one item each is ok if the function applies to arrays with both missing and non missing values? I am just considering whether it's worth having more values in the tested Series' to prove the masks are working as intended.

Reply:
We want to check that the code path we entered before is not entered in the future, doesn't matter how many elements we have, the new code path isn't value dependent.

Reply:
Ok understood. Thanks for the discussion.